base/essentials.jl

```diff
@@ -158,6 +158,22 @@ macro boundscheck(blk)
       $(Expr(:boundscheck,:pop)))
 end
 
+"""
+    @inbounds(blk)
+
+In the example below the bound check of array A is skipped to improve performance.
+```julia
+function sum(A::AbstractArray)
+    r = zero(eltype(A))
+    for i = 1:length(A)
+        @inbounds r += A[i]
+    end
+    return r
+end
+```
+!!! Warning: Using `@inbounds` may return incorrect results/clashes/corruption
```

Comment:
This should be

```
!!! warning

    Using `@inbounds` may return incorrect results/clashes/corruption for out-of-bounds indices.
    The user is responsible for checking it manually.
```

for it to render correctly as a warning block in the docs.